core/src/main/java/com/graphhopper/routing/ev/DecimalEncodedValue.java

```diff
@@ -40,4 +40,6 @@ public interface DecimalEncodedValue extends EncodedValue {
     double getNextStorableValue(double value);
 
     double getSmallestNonZeroValue();
+
+    double getRealMax();
```

Comment:
I called the new method `getLowestUpperBoundInt/Decimal` now. Maybe a bit long, but then again it is a special method and the name tells us what it does. It returns an upper bound, but not just any upper bound, but the lowest (known) one. Ok 'known' is not part of the name... :) But would `getLowestKnownUpperBoundInt/Decimal` be an improvement? I don't really care :)

Maybe we do not even need to rename the other getMin/Max methods then? Or shall we name the new one `getLowestMaxInt/Decimal` and still rename the others to `getMin/MaxStorable`?

Reply:
Internally we have `max_value` and `max_set_value` now. Does this make it clear? Or should we use `max_storable_value` and `max_set_value`, or `max_stored_value`? Or just `max_storable_value` and `max_value`?

But `max_stored_value` is not exactly correct because it is not the value that is stored, but the one before it is shifted by minValue. Also `max_value` might be misleading, because it is not the max_value when the value is later set to a smaller value, but rather the maximum value that was ever **set**... -> `max_set_value` makes sense to me.

Reply:
I'm happy with the current code (`max_value` and `max_set_value` and `getMaxInt()` and `getLowestUpperBoundInt()`) so if you agree I'll merge, but I'm also open to other suggestions.

Reply:
`getMaxSetValueOrMaxDecimal()`? It's long and correct, but make sure to forget about the ugly Decimal suffix for a moment: `getMaxSetValueOrMax()`.